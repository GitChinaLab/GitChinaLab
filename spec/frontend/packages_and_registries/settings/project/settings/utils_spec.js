import {
  formOptionsGenerator,
  optionLabelGenerator,
  olderThanTranslationGenerator,
} from '~/packages_and_registries/settings/project/utils';

describe('Utils', () => {
  describe('optionLabelGenerator', () => {
    it('returns an array with a set label', () => {
      const result = optionLabelGenerator(
        [{ variable: 1 }, { variable: 2 }],
        olderThanTranslationGenerator,
      );
      expect(result).toEqual([
        { key: null, label: '' },
        { variable: 1, label: '1 day' },
        { variable: 2, label: '2 days' },
      ]);
    });
  });

  describe('formOptionsGenerator', () => {
    it('returns an object containing olderThan', () => {
      expect(formOptionsGenerator().olderThan).toBeDefined();
      expect(formOptionsGenerator().olderThan).toMatchSnapshot();
    });

    it('returns an object containing cadence', () => {
      expect(formOptionsGenerator().cadence).toBeDefined();
      expect(formOptionsGenerator().cadence).toMatchSnapshot();
    });

    it('returns an object containing keepN', () => {
      expect(formOptionsGenerator().keepN).toBeDefined();
      expect(formOptionsGenerator().keepN).toMatchSnapshot();
    });
  });
});
