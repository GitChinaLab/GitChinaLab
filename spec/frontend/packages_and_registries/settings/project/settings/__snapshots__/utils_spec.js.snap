// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils formOptionsGenerator returns an object containing cadence 1`] = `
Array [
  Object {
    "default": true,
    "key": "EVERY_DAY",
    "label": "Every day",
  },
  Object {
    "default": false,
    "key": "EVERY_WEEK",
    "label": "Every week",
  },
  Object {
    "default": false,
    "key": "EVERY_TWO_WEEKS",
    "label": "Every two weeks",
  },
  Object {
    "default": false,
    "key": "EVERY_MONTH",
    "label": "Every month",
  },
  Object {
    "default": false,
    "key": "EVERY_THREE_MONTHS",
    "label": "Every three months",
  },
]
`;

exports[`Utils formOptionsGenerator returns an object containing keepN 1`] = `
Array [
  Object {
    "key": null,
    "label": "",
  },
  Object {
    "default": false,
    "key": "ONE_TAG",
    "label": "1 tag per image name",
    "variable": 1,
  },
  Object {
    "default": false,
    "key": "FIVE_TAGS",
    "label": "5 tags per image name",
    "variable": 5,
  },
  Object {
    "default": true,
    "key": "TEN_TAGS",
    "label": "10 tags per image name",
    "variable": 10,
  },
  Object {
    "default": false,
    "key": "TWENTY_FIVE_TAGS",
    "label": "25 tags per image name",
    "variable": 25,
  },
  Object {
    "default": false,
    "key": "FIFTY_TAGS",
    "label": "50 tags per image name",
    "variable": 50,
  },
  Object {
    "default": false,
    "key": "ONE_HUNDRED_TAGS",
    "label": "100 tags per image name",
    "variable": 100,
  },
]
`;

exports[`Utils formOptionsGenerator returns an object containing olderThan 1`] = `
Array [
  Object {
    "key": null,
    "label": "",
  },
  Object {
    "default": false,
    "key": "SEVEN_DAYS",
    "label": "7 days",
    "variable": 7,
  },
  Object {
    "default": false,
    "key": "FOURTEEN_DAYS",
    "label": "14 days",
    "variable": 14,
  },
  Object {
    "default": false,
    "key": "THIRTY_DAYS",
    "label": "30 days",
    "variable": 30,
  },
  Object {
    "default": false,
    "key": "SIXTY_DAYS",
    "label": "60 days",
    "variable": 60,
  },
  Object {
    "default": true,
    "key": "NINETY_DAYS",
    "label": "90 days",
    "variable": 90,
  },
]
`;
